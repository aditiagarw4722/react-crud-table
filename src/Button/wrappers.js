import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-button');

export const Btn = block('button');

export default { Btn };
