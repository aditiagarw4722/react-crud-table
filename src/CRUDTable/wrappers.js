import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table');

export const Table = block('table');

Table.displayName = 'Table';

Table.Row = element('tr', 'row');
Table.Row.displayName = 'Row';

Table.HeaderCell = element('th', 'header-cell');
Table.HeaderCell.displayName = 'HeaderCell';

Table.Header = element('thead', 'header');
Table.Header.displayName = 'Header';

Table.Body = element('tbody', 'body');
Table.Body.displayName = 'Body';

Table.Cell = element('td', 'cell');
Table.Cell.displayName = 'Cell';

Table.CellLabel = element('div', 'cell-label');
Table.CellLabel.displayName = 'CellLabel';

Table.Caption = element('div', 'caption');
Table.Caption.displayName = 'Caption';

export default { Table };
