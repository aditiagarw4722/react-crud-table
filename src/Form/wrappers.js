import { Form as FormikForm } from 'formik';
import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-form');

export const Form = block(FormikForm);

Form.displayName = 'FormikForm';

Form.Label = element('label', 'label');
Form.Label.displayName = 'Label';

Form.Message = element('div', 'message');
Form.Message.displayName = 'Message';

Form.ErrorMessage = element('div', 'error-message');
Form.ErrorMessage.displayName = 'ErrorMessage';

Form.FieldError = element('div', 'field-error');
Form.FieldError.displayName = 'FieldError';

Form.FieldContainer = element('div', 'field-container');
Form.FieldContainer.displayName = 'FieldContainer';

export default {
  Form,
};
