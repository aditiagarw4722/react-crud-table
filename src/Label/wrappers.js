import bcc from 'bem-react-component-creator';

const { block } = bcc('crud-label');

const Label = block('div');
Label.displayName = 'Label';

export default Label;
