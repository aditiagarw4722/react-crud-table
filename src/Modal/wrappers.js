import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-modal-wrapper');

export const Container = block('div');
Container.BG = element('div', 'background');
Container.Modal = element('div', 'modal');
Container.Title = element('h3', 'title');

export default {
  Container,
};
