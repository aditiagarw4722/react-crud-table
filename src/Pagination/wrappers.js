import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table-pagination');

export const Pagination = block('div');

Pagination.Link = element('button', 'link');
Pagination.Prev = element('button', 'previous');
Pagination.Next = element('button', 'next');

export default { Pagination };
