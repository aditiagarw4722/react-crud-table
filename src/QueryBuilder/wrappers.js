import bcc from 'bem-react-component-creator';

const { block, element } = bcc('crud-table-query-builder');

export const Container = block('div');

Container.displayName = 'Container';

export const RuleBuilder = element('div', 'search-rule-builder');

RuleBuilder.displayName = 'RuleBuilder';

export default { Container, RuleBuilder };
