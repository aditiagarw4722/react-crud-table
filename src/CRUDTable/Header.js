import React from 'react';
import PropTypes from 'prop-types';

import { Table } from './wrappers';
import { chevron } from './helpers';

const Header = ({
  fields,
  sort,
  onClick,
  forms,
  actionsLabel,
}) => (
  <Table.Header>
    <Table.Row>
      {fields.map((field) => (
        <Table.HeaderCell
          onClick={() => {
            if (field.sortable) {
              onClick(field.name, sort.direction);
            }
          }}
        >
          {field.label}
          {' '}
          {sort.field === field.name && chevron(sort.direction)}
        </Table.HeaderCell>
      ))}
      {(forms.delete || forms.update) && (
      <Table.HeaderCell>
        {actionsLabel}
      </Table.HeaderCell>
      )}
    </Table.Row>
  </Table.Header>
);

Header.propTypes = {
  onClick: PropTypes.func,
  actionsLabel: PropTypes.node,
  forms: PropTypes.instanceOf(Object),
  fields: PropTypes.instanceOf(Array),
  sort: PropTypes.instanceOf(Array).isRequired,
};

Header.defaultProps = {
  fields: [],
  onClick: () => {},
  actionsLabel: '',
  forms: {},
};

export default Header;
