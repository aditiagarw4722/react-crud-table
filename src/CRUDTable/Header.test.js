import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { create } from 'react-test-renderer';

import { Table } from './wrappers';
import Header from './Header';

const renderer = new ShallowRenderer();

describe('Header', () => {
  let props;

  beforeEach(() => {
    props = {
      trigger: 'Open',
      title: 'Test title',
      onClick: jest.fn(),
      actionsLabel: 'Act',
      fields: [
        {
          name: 'name',
          label: 'Name',
          sortable: true,
        },
        {
          name: 'age',
          label: 'Age',
          sortable: false,
        },
      ],
      sort: {
        field: 'name',
        direction: 'ascending',
      },
    };
  });

  it('renders as expected with the given props', () => {
    const result = renderer.render(<Header {...props} />);
    expect(result).toMatchSnapshot();
  });

  it('renders as expected with the given props (descending sort)', () => {
    props.sort.direction = 'descending';
    const result = renderer.render(<Header {...props} />);
    expect(result).toMatchSnapshot();
  });

  it('notifies when opened', () => {
    const result = create(<Header {...props} />);

    const cells = result.root.findAllByType(Table.HeaderCell);

    cells[0].props.onClick();
    cells[1].props.onClick();

    expect(props.onClick).toHaveBeenCalledTimes(1);
    expect(props.onClick).toHaveBeenCalledWith('name', 'ascending');
  });
});
