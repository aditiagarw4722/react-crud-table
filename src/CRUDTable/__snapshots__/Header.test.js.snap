// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders as expected with the given props (descending sort) 1`] = `
<Header>
  <Row
    modifiers="fields"
  >
    <HeaderCell
      onClick={[Function]}
    >
      Name
       
      <DownArrow />
    </HeaderCell>
    <HeaderCell
      onClick={[Function]}
    >
      Age
       
    </HeaderCell>
    <HeaderCell>
      Act
    </HeaderCell>
  </Row>
  <Row
    modifiers="sorter"
  >
    <HeaderCell
      colSpan={3}
    >
      Sort by: 
      <br />
      <Select
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "name",
              "text": "Name",
              "value": "name",
            },
          ]
        }
        placeholder="Select field"
        value="name"
      />
       
      <Select
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "DESCENDING",
              "text": "descending",
              "value": "descending",
            },
            Object {
              "key": "ASCENDING",
              "text": "ascending",
              "value": "ascending",
            },
          ]
        }
        placeholder="Select direction"
        value="descending"
      />
    </HeaderCell>
  </Row>
</Header>
`;

exports[`Header renders as expected with the given props 1`] = `
<Header>
  <Row
    modifiers="fields"
  >
    <HeaderCell
      onClick={[Function]}
    >
      Name
       
      <UpArrow />
    </HeaderCell>
    <HeaderCell
      onClick={[Function]}
    >
      Age
       
    </HeaderCell>
    <HeaderCell>
      Act
    </HeaderCell>
  </Row>
  <Row
    modifiers="sorter"
  >
    <HeaderCell
      colSpan={3}
    >
      Sort by: 
      <br />
      <Select
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "name",
              "text": "Name",
              "value": "name",
            },
          ]
        }
        placeholder="Select field"
        value="name"
      />
       
      <Select
        onChange={[Function]}
        options={
          Array [
            Object {
              "key": "DESCENDING",
              "text": "descending",
              "value": "descending",
            },
            Object {
              "key": "ASCENDING",
              "text": "ascending",
              "value": "ascending",
            },
          ]
        }
        placeholder="Select direction"
        value="ascending"
      />
    </HeaderCell>
  </Row>
</Header>
`;
